### ISSUE TITLE
Node.js was not found. Make sure to install Node.js.

### ISSUE THREAD

**Issue by webmigratesmj** [software-mansion/react-native-ide#411]
**Description*

**Environment**
- Version of currently used React Native IDE (Extensions tab -> click on React Native IDE extension -> version is located next to the extension name)
- Version of React Native/Expo used in the project.
- Any other versions of tools related to the problem.

**Comment by webmigratesmj** [comment]
I also tried this asdf global nodejs 20.11.0 but still not able to resolve

**Comment by jakub-gonet** [comment]
Do you open VSCode from terminal or applications? If the latter, asdf environment may not load so we won't see node binary.

**Comment by webmigratesmj** [comment]
> Do you open VSCode from terminal or applications? If the latter, asdf environment may not load so we won't see node binary.

I have tried both

**Comment by iangregsondev** [comment]
I have the same issue, i am using nvm to manage my node versions

Anybody manage to fix it ?

**Comment by jakub-gonet** [comment]
@iangregsondev what version of IDE are you using? On macos? Since few versions we load the PATH so we should have node available regardless of how it was installed.

**Comment by iangregsondev** [comment]
Hi @jakub-gonet ,

I am using Radon IDE V.0.0.19 - I just installed it.

Vscode - Version: 1.94.0

Macos - yes, 15.0.1 - latest.

Node - I was using the the latest but I went back a few versions to see if I could get it to work.

 node -v
v20.11.0

It also doesn't detect Xcode either, but I know xcode and command line tools are working

![CleanShot 2024-10-05 at 11 49 29@2x](https://github.com/user-attachments/assets/7a877d0a-79ad-4d42-b9b6-31b079b35e55)

**Comment by iangregsondev** [comment]
This same project, if I do a npx expo start

Everything works as expected, no errors.

**Comment by jakub-gonet** [comment]
Can you please attach logs from IDE after it starts? View -> output -> select Radon IDE in tab on the right.

**Comment by iangregsondev** [comment]
Sure,

https://pastebin.com/vhX4rtKY

**Comment by BMR11** [comment]
I am facing the same issue

**Comment by jakub-gonet** [comment]
@iangregsondev you're getting `ENAMETOOLONG: npm --version`. What the output of `which node` when you run it in the terminal?

**Comment by jakub-gonet** [comment]
Are you using windows? You didn't fill out the issue template.

**Comment by iangregsondev** [comment]
@jakub-gonet 

npm --version
10.8.3

The npm is dependent on my node installed via NVM

Here is it 

which npm
/Users/ian/.config/nvm/versions/node/v22.9.0/bin/npm

Anything else you need, please let me know.

Thanks in advance

**Comment by germanolira** [comment]
Same here, even with the node_modules installed, it still shows the error "Node modules are not installed"

And when you click to open the IDE, it opens but doesn't show anything, you have to open and close it several times for the emulator to appear

Version: 0.42.2
VSCode Version: 1.93.1
Node.js: 20.15.1
MacOS 14.2

![Captura de Tela 2024-10-12 às 16 21 52](https://github.com/user-attachments/assets/c8502018-dd2c-4f3e-8a55-df504ac83d72)

**Comment by germanolira** [comment]
I downgraded from version 0.20 to 0.19 and switched from bun to npm, and it worked perfectly

**Comment by ecexplorer** [comment]
I'm running into the same issue, using https://www.jetify.com/devbox for node installation local to the project path. Would be nice to have a config option which points to node location (similar to .xcode.env)

**Comment by kmagiera** [comment]
can you specify whether you start vscode from command line or open it using some application launcher? This may be relevant as depending on the way you launch vscode, it may have different environment variables.

**Comment by iangregsondev** [comment]
Not sure if that is a question to me @kmagiera , but I pretty much always to a 

```
code .
```

from the directory I want to work in

**Comment by ecexplorer** [comment]
@kmagiera This solved this issue for me, I just had to launch VSCode from my project path that had access to the node location.
Also, I realized Radon wasn't the only one limited by this setup and what I assume is a node path for the _whole_ of VSCode (I also found that my Jest extension wasn't finding node until launching from the project path as well).

**Comment by kmagiera** [comment]
yes, it is pretty tricky to get it right with all different types of tool managers. We actually support mise and asdf already. The way they work is by updating the PATH variable depending on the current location. What we do is we start shell that enters the workspace location and the prints PATH, we then use that PATH variables when launching subprocesses including node.

Here's the relevant code in Radon IDE that handles that: https://github.com/software-mansion/radon-ide/blob/644656f5c384d282dcdefb219bc84dba003e28e7/packages/vscode-extension/src/utilities/subprocess.ts#L19

Perhaps devbox works in a differently. If you have any details or would like to
